docs/guides/custom-http-client.mdx

```diff
@@ -0,0 +1,48 @@
+---
+id: custom-http-client
+title: Using a custom HTTP client (Experimental)
+description: Use a custom HTTP client for `sendRequest` and plain-HTTP crawling
+---
+
+import ApiLink from '@site/src/components/ApiLink';
+
+The <ApiLink to="core/basic/BasicCrawler">`BasicCrawler`</ApiLink> class allows you to configure the HTTP client implementation using the `httpClient` constructor option. This might be useful for testing or if you need to swap out the default implementation based on `got-scraping` for something else, such as `curl-impersonate` or `axios`.
+
+The HTTP client implementation needs to conform to the `BaseHttpClient` interface:
+
+```ts
+import {
+	BaseHttpClient,
```

Comment:
mixed tabs vs spaces

Reply:
Yeah, I'm going to pull it out into a file and run it through a formatter.